drivers/FileBase.cpp

```diff
@@ -52,8 +53,8 @@ FileBase::~FileBase() {
     _mutex->unlock();
 
     if (getPathType() == FilePathType) {
-        extern void remove_filehandle(FileLike *file);
-        remove_filehandle(static_cast<FileLike*>(this));
+        extern void remove_filehandle(FileHandle *file);
```

Comment:
Oh, we actually have retarget.h now, perhaps system functions should go there? retarget.h sounds like a reasonable location for fdopen at least.